pre_commit/commands/run.py

```diff
@@ -245,6 +245,15 @@ def _run_hooks(
         environ: EnvironT,
 ) -> int:
     """Actually run the hooks."""
+
+    # Skip running any hooks if _PRE_COMMIT_SKIP_POST_CHECKOUT is set (#1418)
+    # and this is an execution for post-checkout hooks.
+    if (
+        args.hook_stage == 'post-checkout' and
+        os.getenv('_PRE_COMMIT_SKIP_POST_CHECKOUT') is not None
+    ):
+        return False
```

Comment:
while `False` is a valid integer in python, the intention here is to be the program exit code so it would be more expected to `return 0`